packages/code-du-travail-frontend/src/outils/indemniteLicenciement/Wizard.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { Form } from "react-final-form";
+import arrayMutators from "final-form-arrays";
+import { StepItems } from "./StepItems";
+import { PrevNextBar } from "./PrevNextBar";
+
+function Wizard({
+  initialData = {},
+  initialStep = 0,
+  steps,
+  onSubmit,
+  onUpdate,
+  rules
+}) {
+  const [initialValues, setValues] = useState(initialData);
+  const [page, setPage] = useState(initialStep);
```

Comment:
Je ne comprends pas pourquoi est-ce qu'on a deux useState ici. Je pensais que le state était géré via le reducer dans l'index qui semble avoir un prop page ?
Je pensais que le wizard aurait simplement dispatch et qu'il ferait le dispatch nécessaire avec la bonne action pour changer de page, values & co. Mais là on a deux components qui gèrent des choses similaires mais différentes et qui se communiquent ça via des props ? Je ne suis pas certain de saisir le pourquoi du comment.
J'ai peur que le initialStep & initialData, ne soient pas si initial que ça en fait. Si j'ai bien compris là, à chaque fois qu'un dispatch de UPDATE_DATA sera fait, les initialProps vont changer, tout sera recalculé etc dans le wizard non ? Ce qui fait qu'elle ne sont pas si initial que ça finalement ?

La j'ai du mal à comprendre quel state sert à quoi et qui fait quoi exactemennt ?

Reply:
C’etait L’idée au départ mais je me suis rendu compte que mette à jour le state a chaque update mais le form ayant lui aussi un state interne et le fait n’utilise pas le state du composant implique de clarifier un peu les responsabilités.
On pourrait sortir la partie data est conserver seulement la partie step

Reply:
Je ne suis pas certain d'avoir saisi mais il me semble que changer la prop dans initialValues à postériori n'impacte pas le formulaire. C'est une prop "one shot" qui n'est utilisé qu'à l'instantiation. (C'est le même gars qui a crée redux form donc je retrouve quelques similarités :D )

Reply:
> Il me semble que c'est une prop "one shot" qui n'est utilisée qu'à l'instantiation

ca ne l'ai pas justement https://codesandbox.io/s/react-final-form-loading-and-initializing-eb1wr

Reply:
Ah ou effectivement elle ne l'est pas du tout, my bad :D Du coup tu peux mettre à jour programmatiquement la data simplement en changeant le initialValues ? Plutôt cool, ca change de la version avec redux on dirait bien